roles/test-idp/tasks/main.yml

```diff
@@ -0,0 +1,92 @@
+---
+#########################################################
+## Demop app
+#########################################################
+- name: Create html dir
+  file:
+    path: "{{testidp_html_dir}}"
+    state: "directory"
+    mode: "0755"
+
+#########################################################
+## SAML IdP
+#########################################################
+- include_role:
+    name: "simplesamlphp"
+    public: true
+  vars:
+    simplesaml_name: "{{testidp_name}}"
+    simplesaml_user: "{{testidp_user}}"
+    simplesaml_group: "{{testidp_group}}"
+    simplesaml_project_dir: "{{testidp_dir}}"
+
+- name: SimpleSaml configuration (templates)
+  template:
+    src: "{{item}}.j2"
+    dest: "{{ simplesaml_project_dir }}/config/{{item}}"
+  with_items:
+    - "saml20-idp-hosted.php"
+
+- name: SimpleSaml configuration (copy)
+  copy:
+    src: "{{item}}"
+    dest: "{{ simplesaml_project_dir }}/config/{{item}}"
+  with_items:
+    - "saml20-idp-remote.php"
+    - "saml20-sp-remote.php"
+
+- name: Link SimpleSaml configuration
+  file:
+    src: "{{ simplesaml_project_dir }}/config/{{item.key}}"
+    path: "{{ simplesaml_project_dir }}/simplesaml/{{item.value}}/{{item.key}}"
+    state: "link"
+    force: true
+  with_dict:
+    saml20-idp-hosted.php: "metadata"
+    saml20-idp-remote.php: "metadata"
+    saml20-sp-remote.php: "metadata"
+
+- name: Generate a SAML signing certificate
+  command:
+    cmd: >
+      openssl req -x509 -newkey rsa:2048 -sha256 -days 36500 -nodes
+              -keyout saml.key -out saml.pem
+              -subj '/CN=simpelsamlphp/'
+    chdir: "{{simplesaml_cert_dir}}"
+    creates: "{{simplesaml_cert_dir}}/saml.pem"
+
+- name: Set permissions for keys
+  file:
+    path: "{{simplesaml_cert_dir}}/saml.key"
+    mode: "0640"
+    group: "{{testidp_group}}"
+
+#########################################################
+## nginx
+#########################################################
+- include_role:
+    name: "nginx"
+
+- include_role:
+    name: "letsencrypt"
+    public: true
+  vars:
+    letsencrypt_staging: false
+    letsencrypt_hosts: [ "test-idp.sram.surf.nl" ]
+
+- name: Install nginx config
+  template:
+    src: "nginx.conf.j2"
+    dest: "/etc/nginx/sites-enabled/01-sram-testidp.conf"
+  notify: "restart nginx"
+
+- name: Restart nginx after certificate rollover
+  copy:
+    content: |
+      #!/bin/sh
+      echo "restarting nginx"
+      /usr/bin/systemctl reload nginx.service
+      exit 0
+    dest: "{{letsencrypt_hooks}}/sram-nginx.sh"
+    mode: "0755"
+
```

Comment:
Hoort deze laatste task niet thuis in de nginx task?  Als je hem hier zet gaan er potentieel rollen met elkaar interfereren als ze dit allebei doen.

Reply:
Ik heb dit een-op-een overgenomen van demo-sp, dus als het hier niet hoort, moeten we daar ook naar kijken?
https://github.com/SURFscz/SRAM-deploy/blob/main/roles/demo-sp/tasks/main.yml#L126

Reply:
ok, heb het vanuit beide rollen verplaatst